ext/TrixiNLsolveExt.jl

```diff
@@ -0,0 +1,70 @@
+# Package extension for adding Convex-based features to Trixi.jl
+module TrixiNLsolveExt
+
+# Required for coefficient optimization in P-ERK scheme integrators
+if isdefined(Base, :get_extension)
+    using NLsolve: nlsolve
+else
+    # Until Julia v1.9 is the minimum required version for Trixi.jl, we still support Requires.jl
+    using ..NLsolve: nlsolve
+end
+
+using Random: seed!
+
+# Use functions that are to be extended and additional symbols that are not exported
+using Trixi: Trixi, solve_a_unknown!, PairedExplicitRK3_butcher_tableau_objective_function,
+             @muladd
+
+# By default, Julia/LLVM does not use fused multiply-add operations (FMAs).
+# Since these FMAs can increase the performance of many numerical algorithms,
+# we need to opt-in explicitly.
+# See https://ranocha.de/blog/Optimizing_EC_Trixi for further details.
+@muladd begin
+#! format: noindent
+
+# Find the values of the a_{i, i-1} in the Butcher tableau matrix A by solving a system of
+# non-linear equations that arise from the relation of the stability polynomial to the Butcher tableau.
+# For details, see Proposition 3.2, Equation (3.3) from 
+# Hairer, Wanner: Solving Ordinary Differential Equations 2
+function Trixi.solve_a_unknown!(a_unknown, num_stages, monomial_coeffs, c_s2, c;
+                                verbose)
+    is_sol_valid = false
+
+    # Define the objective_function
+    function objective_function(x)
+        return Trixi.PairedExplicitRK3_butcher_tableau_objective_function(x, num_stages,
+                                                                          num_stages,
+                                                                          monomial_coeffs,
+                                                                          c_s2)
+    end
+
+    seed!(5555)
+
+    while !is_sol_valid
+        # Due to the nature of the nonlinear solver, different initial guesses can lead to 
+        # small numerical differences in the solution.
+        # To ensure consistency and reproducibility of results across runs, we use 
+        # a seeded random initial guess.
+        x0 = 0.1 .* rand(num_stages - 2)
```

Comment:
Have we tried setting the initial guess to 0 after your bug-fixes? Maybe this works as well and we can get rid of all the random stuff.

Reply:
I tried removing randomizing initial guess and not using this !is_sol_valid while loop. It works for lower numbers of stages. However, for `s = 16`, we sometimes get negative coefficients, hence the loop has to stay and the initial guesses have to be changed via randomization.

Reply:
True, it is not clear that this always finds a solution. We should maybe safe-guard this with a number of maximum iterations, that is a good idea :+1:

Reply:
Copy that!